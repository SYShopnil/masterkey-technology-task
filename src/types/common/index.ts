export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EAuth {
  AuthTokenCookieName = "auth",
}

export enum EDataTestId {
  cLoginFormWithSubmit = "cLoginFormWithSubmit",
  cLogoutContainer = "cLogoutContainer",
  SHeaderMain = "SHeaderMain",
  SProfile = "SProfile",
  CAddNewText = "CAddNewText",
  CInputCheckBoxContainer = "CInputCheckBoxContainer",
  SShowAllTextBox = "SShowAllTextBox",
  STextBox = "STextBox",
  CButton = "CButton",
  CLogout = "CLogout",
  CModal = "CModal",
  SIconWithMessage = "SIconWithMessage",
  SLoading = "SLoading",
  STextCardSkelton = "STextCardSkelton",
  ProfilePage = "ProfilePage",
}
