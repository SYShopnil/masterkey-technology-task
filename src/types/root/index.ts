export type { IIconStore } from "./_icon";
export { IconName as IconEnums } from "./_icon";
