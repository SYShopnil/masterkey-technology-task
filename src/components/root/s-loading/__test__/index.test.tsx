import { render, screen } from "@testing-library/react";
import { EDataTestId } from "@src/types/common";
import { SLoading } from "..";

describe("When loading server component has rendered", () => {
  it("Expect render successfully", () => {
    render(<SLoading text="Loading..." />);
    const myElement = screen.getByRole(EDataTestId.SLoading);
    expect(myElement).toMatchSnapshot();
  });
});
