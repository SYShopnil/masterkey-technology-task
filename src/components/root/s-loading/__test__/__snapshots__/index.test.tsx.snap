// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When loading server component has rendered Expect render successfully 1`] = `
<div
  class="flex justify-center items-center h-[50vh]"
  role="SLoading"
>
  <p
    class="text-center font-extrabold text-2xl text-[#7F4D4F]"
  >
    Loading...
  </p>
</div>
`;
